identity-apps-core/apps/authentication-portal/src/main/webapp/org_handle.jsp

```diff
@@ -0,0 +1,214 @@
+
+    String errorMessage = AuthenticationEndpointUtil.i18n(resourceBundle, "error.retry");
+    String authenticationFailed = "false";
+
+    // Log the actual error for localized error fallbacks
+    boolean isErrorFallbackLocale = !userLocale.toLanguageTag().equals("en_US");
+
+    if (Boolean.parseBoolean(request.getParameter(Constants.AUTH_FAILURE))) {
+        authenticationFailed = "true";
+
+        if (request.getParameter(Constants.AUTH_FAILURE_MSG) != null) {
+            errorMessage = request.getParameter(Constants.AUTH_FAILURE_MSG);
+
+            if (errorMessage.equalsIgnoreCase("authentication.fail.message")) {
+                errorMessage = AuthenticationEndpointUtil.i18n(resourceBundle, "error.retry");
+            } else if (errorMessage.equalsIgnoreCase("Invalid Organization Handle")) {
+                errorMessage = AuthenticationEndpointUtil.i18n(resourceBundle, "invalid.organization.handle");
+            } else if (isErrorFallbackLocale) {
+                errorMessage = AuthenticationEndpointUtil.i18n(resourceBundle,"error.retry");
+            }
+        }
+    }
+    boolean isOrgDiscoveryEnabled = Boolean.parseBoolean(request.getParameter("orgDiscoveryEnabled"));
+%>
+
+<%-- Data for the layout from the page --%>
+<%
+    layoutData.put("containerSize", "medium");
+%>
+
+<html lang="en-US">
+    <head>
+        <%-- header --%>
+        <%
+            File headerFile = new File(getServletContext().getRealPath("extensions/header.jsp"));
+            if (headerFile.exists()) {
+        %>
+                <jsp:include page="extensions/header.jsp"/>
+        <%
+            } else {
+        %>
+                <jsp:include page="includes/header.jsp"/>
+        <%
+            }
+        %>
+    </head>
+
+    <body class="login-portal layout authentication-portal-layout">
+        <layout:main layoutName="<%= layout %>" layoutFileRelativePath="<%= layoutFileRelativePath %>" data="<%= layoutData %>" >
+            <layout:component componentName="ProductHeader">
+                <%-- product-title --%>
+                <%
+                    File productTitleFile = new File(getServletContext().getRealPath("extensions/product-title.jsp"));
+                    if (productTitleFile.exists()) {
+                %>
+                        <jsp:include page="extensions/product-title.jsp"/>
+                <%
+                    } else {
+                %>
+                        <jsp:include page="includes/product-title.jsp"/>
+                <%
+                    }
+                %>
+            </layout:component>
+            <layout:component componentName="MainSection">
+                <div class="ui segment">
+                    <%-- page content --%>
+                    <h2><%=AuthenticationEndpointUtil.i18n(resourceBundle, "sign.in.with")%> <%= StringUtils.isNotBlank(idp) ? Encode.forHtmlContent(idp) : AuthenticationEndpointUtil.i18n(resourceBundle, "organization.login") %></h2>
```

Comment:
will there be a way to get the `idp` value other than using request parameters? We already are facing an issue due to a similar implementation. Let's try to embed this value server side if possible.

cc: @sumedhe